test/scylla-cli/run

```diff
@@ -0,0 +1,63 @@
+#! /bin/bash -e
+
+function usage () {
+    echo "Usage $(basename $0) <path/to/scylla>" 1>&2
+    exit 1
+}
+
+scylla="$(realpath ${SCYLLA-$1})"
+[ -x "$scylla" ] || usage
+
+here=$(dirname $(readlink -e "$0"))
+src=$(realpath -e "$here"/../..)
+scylla_cli=$(realpath -e "$src"/tools/scylla-cli/scylla-cli.py)
+
+_WORKDIR="$(readlink -e ${TMPDIR-/tmp})"/scylla-cli-$$
+scylla_link="$_WORKDIR"/test-cli-scylla
+scylla_log="$_WORKDIR"/test-cli-scylla.log
+scylla_status_file="$_WORKDIR"/test-cli-scylla.status
+mkdir "$_WORKDIR"
+
+code=0
+scylla_code=0
+cleanup() {
+    pkill --parent $$ || echo "Could not pkill $scylla_link"
+    rm -rf --preserve-root "$_WORKDIR"
+    if [ $code != 0 ]; then
+        echo "test failed with code $code" 1>&2
+        exit $code
+    elif [ $scylla_code != 0 ]; then
+        echo "scylla exited with code $scylla_code" 1>&2
+        exit 1
+    fi
+    exit 0
+}
+trap 'cleanup' EXIT
+
+# for easier pkill
+ln -s "$scylla" "$scylla_link"
+
+# see explanation of SCYLLA_IP in ../alternator/run
+scylla_ip=127.1.$(($$ >> 8 & 255)).$(($$ & 255))
+
+{ "$scylla_link" --developer-mode=true -W "$_WORKDIR" \
+    -c2 -m256M \
+    --api-address $scylla_ip \
+    --rpc-address $scylla_ip \
+    --listen-address $scylla_ip \
+    --prometheus-address $scylla_ip \
+    --seed-provider-parameters seeds=$scylla_ip \
+        >& "$scylla_log"; echo $? > "$scylla_status_file"; } &
+
+while ! grep -q "Scylla API server listening" "$scylla_log" && ! test -f "$scylla_status_file"; do
+    sleep 1;
+done
```

Comment:
This loop needs some sort of timeout and failure...
Also, if I were you I would check the actual  REST API in an actual request, instead of grepping the log.

Reply:
The exit condition is the existence of `scylla_status_file`, meaning scylla exited.
We can have a timeout too.

Testing the API happens right after this loop exits.  We can use curl to sample the api to see if it's ready instead of grepping the log.

Reply:
I see. I missed that "scylla_status_file" trick. However, there is still the risk that if Scylla hangs before printing this "server listening" message, or if one day somebody changes this message, this test will not only start to fail - it will fail by hanging forever, which isn't pretty.

Right, I meant you can use curl (for example) to check that the REST API is up. Not using the new "scylla-cli" but directly. The benefit of doing this over log grepping is that it isn't sensitive to log messages which one day might change.